web_infrastructure/jenkins_plugin.py

```diff
@@ -567,7 +567,7 @@ def _download_updates(self):
             updates_file = tempfile.mkstemp()
 
             try:
-                fd = open(updates_file, 'wb')
+                fd = open(updates_file[1], 'wb')
```

Comment:
This change _did_ get rid of one of the errors reported in #3391, but I had to make a similar edit to the tmp file opening. When I did this, I got _different_ errors.
```json

{"failed": true, "item": "analysis-core", "msg": "Could not replace file: (20, '/tmp/tmpd58_hC') to /var/lib/jenkins/plugins/analysis-core.jpi: [Errno 2] No such file or directory"}
```

So, this little experiment of mine didn't quite work.